app/models/restaurant.rb

```diff
@@ -15,8 +15,14 @@
 
 class Restaurant < ApplicationRecord
 
-  validates :name, :address, :accepts_10bis, :max_delivery_time_minutes, presence: true
+  validates_presence_of :name, :address, :max_delivery_time_minutes
+  validates_inclusion_of :accepts_10bis, in: [true, false]
 
   belongs_to :cuisine
+  has_many :reviews
 
+
+  def calc_rating
+    self.rating = reviews.empty? ? nil: reviews.average(:rating).round
```

Comment:
1) Why are you rounding? To me this sounds like more of a UI concern
2) An exercise for your mind: How will that perform when there are a lot of reviews? How can you address that?

Reply:
👍 
`reviews` table should be incredibly huge for bad performance (as `average` performs an SQL query) so for now I'll index `(:restaurant_id, :rating)`, but if we expect the table to be really big I would calculate differently.
Thanks 😊